viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -86,19 +100,28 @@ export class FilterTree extends React.Component<FilterTreeProps, FilterTreeState
         let checkedIds: number[] = [];
         const checkedNode = this.getNode(this.props.nodes, id);
         if (checkedNode) {
-            if (checkedNode.children.length) {
-                const childrenIds = this.getAllChildrenIds(checkedNode, []);
-                const visibleChildrenIds = childrenIds.filter((childId: number) =>this.getNode(this.state.filteredNodes, childId) !== undefined);
-                if (this.isNodeChecked(id)) {
-                    checkedIds = checkedIds.concat(visibleChildrenIds);
-                } else {
+            const childrenIds = this.getAllChildrenIds(checkedNode, []);
+            const visibleChildrenIds = childrenIds.filter((childId: number) =>this.getNode(this.state.filteredNodes, childId) !== undefined);
```

Comment:
This line could be extracted to getVisibleChildren(). Line 150 has a similar call